packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -62,7 +62,10 @@ export function assetImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
           }
 
           const url = rawUrl.slice(1, -1)
-          const file = path.resolve(path.dirname(id), url)
+          const resolved = await this.resolve(url, id)
+          const file = resolved
+            ? resolved.id
+            : path.resolve(path.dirname(id), url)
```

Comment:
Second @poyoho's comment that we could evaluate `new URL('@some/package')` without `import.meta.url` too.

And with that in mind, we could handle this part of the code differently. If the user adds a `import.meta.url`, we'll use the `path.resolve(path.dirname(id), url)` method; If they don't have `import.meta.url`, we use the `this.resolve(url, id)` method.

Reply:
oh, this is right. `new URL('/hello.js')` is only appear on file in `publish` dirs?

Reply:
I think url that starts with `/` would be resolved from the project root like [glob imports](https://vitejs.dev/guide/features.html#glob-import)

Reply:
I forgot to push up changes that I had amended locally. If the URL starts with `/`, the existing behavior is to resolve a file from the public base dir. I've added a test for this scenario